hw/ip/rv_core_ibex/doc/boot-rom-patching.md

```diff
@@ -175,51 +175,67 @@ In this case, the dedicated SRAM can be locked after being loaded so that the pa
 
 A patch is a short sequence of instructions that replace some from the original code.
 The replacement is performed by re-routing instruction fetches to fetch from alternative addresses.
-This re-routing is handled by a block that sits between the Ibex core and TL-UL crossbar.
-The block is configured by 32 *patch match registers*.
+This re-routing is handled by a logic block that sits between the Ibex core and TL-UL crossbar.
+The block is configured by 32 sets of 4 match registers:
+- [`IBUS_ADDR_MATCHING`](../data/rv_core_ibex.hjson#ibus_addr_matching_0)
+- [`IBUS_ADDR_EN`](../data/rv_core_ibex.hjson#ibus_addr_en_0)
+- [`IBUS_REMAP_ADDR`](../data/rv_core_ibex.hjson#ibus_remap_addr_0).
+- [`IBUS_REGWEN`](../data/rv_core_ibex.hjson#ibus_regwen_0).
 
-A patch match register is structured as follows.
-The patch applies to a small region of code (4, 8, 16 or 32 bytes), starting at a given address.
-This region is encoded in a field called `M_BASE`.
-The destination of the redirection is given by its address, encoded in a field called `R_BASE`.
+The patch loader is responsible for configuring those registers for each patch in the patch table.
+Each entry in the OTP ROM patch table is composed of two 32 bit words with the following layout:
 
-- If the source region has address `yyy...yyy`, the `M_BASE` field is:
+![Patch match register layout](patch-match.svg)
+
+- The first 32 bits describe the patch match configuration:
+  - `M_BASE` is the base address of the original code in ROM to be patched.
+  - `P_SIZE` is the size, in `DWORD`s, of the code region to be patched.
+    ROM patches are small code regions that can be 4, 8 16 or 32 bytes long, and thus the only valid values for `P_SIZE` are 1, 2, 4 and 8.
```

Comment:
I agree! But we can probably drop everything before "the only" without changing the meaning. I wonder whether this should be something like:
```
The only valid values for P_SIZE are 1, 2, 4 and 8.
Any other value would describe a code region of an unsupported size.
This makes the patch invalid and the patch loader will not enable it.
```

Reply:
Good suggestion, thanks. I used it verbatim.